waiter/test/waiter/scheduler/kubernetes_test.clj

```diff
@@ -875,4 +881,17 @@
           (is (thrown? Throwable (kubernetes-scheduler (assoc base-config :pod-base-port 1234567890))))))
 
       (testing "should work with valid configuration"
-        (is (instance? KubernetesScheduler (kubernetes-scheduler base-config)))))))
+        (is (instance? KubernetesScheduler (kubernetes-scheduler base-config))))
+
+      (testing "periodic auth-refresh task"
+        (let [kill-task-fn (atom (constantly nil))
+              orig-start-auth-renewer start-auth-renewer
+              secret-value "secret-value"]
+          (try
+            (with-redefs [start-auth-renewer #(reset! kill-task-fn (apply orig-start-auth-renewer %&))]
+              (is (instance? KubernetesScheduler (kubernetes-scheduler (assoc base-config :authentication {:refresh-delay-mins 1
+                                                                                                           :refresh-fn `test-auth-refresher
```

Comment:
It'd be nice to settle on a convention for the entry name for the function symbol, e.g. `:fn`, in all the places in the config where we use this pattern. This entry can then be placed as the first entry of the config map (like we do for `kind` or `factory-fn`).

Reply:
The problem with `:fn` is that it becomes `fn` when destructuring, and we don't want to shadow a core symbol like that. (I tried that before with the old `:edn-params` option—it was really ugly.)

> This entry can then be placed as the first entry of the config map

Wait... You're saying that you *don't* want it sorted lexicographically?! 😮

Reply:
If we decide to rename all the fn-symbol entries to use a common keyword, that would need to go in a separate PR. If you feel strongly about that, you should probably open a new issue for that task.

Reply:
I think we should settle on a name and use it, `:fn` may not be that name due to the issue you mentioned.

I searched `config-full.edn` and didn't find another place where we have a non-`factory-fn`. Since, this is the first PR that's introducing this name, it makes sense for this PR to also introduce that standard.

Reply:
Renamed `:refresh-fn` to `:action-fn` per our offline discussion.